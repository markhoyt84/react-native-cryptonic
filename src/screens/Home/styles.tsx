import { StyleSheet } from "react-native";

const styles: any = StyleSheet.create({
    container: {
        backgroundColor: "#FBFAFA",
    },
    loadingIndicator: {
        flex: 1,
        marginTop: 100
    }
});
export default styles;
