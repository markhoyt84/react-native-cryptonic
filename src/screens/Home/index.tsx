import * as React from "react";
import {
    Container,
    Header,
    Title,
    Content,
    Body,
    List
} from "native-base";
import { ActivityIndicator } from 'react-native';

import styles from "./styles";
import AssetRow from "../../components/AssetRow";
export interface Props {
    assets: any;
    viewingCurrentAssetTrades: string;
    recentTradeData: [];
    loadRecentTradeDataForKeyPair: Function;
    loadingAssets: boolean;
}
export interface State {}
class Home extends React.Component<Props, State> {

    _loadRecentTradeData = (forKeyPairName:string) => {
        this.props.loadRecentTradeDataForKeyPair(forKeyPairName);
    };

    render() {
        const assetKeys = Object.keys(this.props.assets);
        return (
            <Container style={styles.container}>
                <Header style={styles.header}>
                    <Body>
                    <Title>Cryptonic Markets</Title>
                    </Body>
                </Header>
                <Content>
                    { this.props.loadingAssets &&
                    <ActivityIndicator
                        size='large'
                        style={styles.loadingIndicator}
                    />
                    }
                    <List>
                        {assetKeys.map((item, i) => (
                            <AssetRow
                                asset={this.props.assets[item]}
                                key={item}
                                loadRecentTradeData={this._loadRecentTradeData}
                                recentTradeData={this.props.recentTradeData}
                                viewingCurrentAssetTrades={this.props.viewingCurrentAssetTrades}
                            />
                        ))}
                    </List>
                </Content>
            </Container>
        );
    }
}

export default Home;
