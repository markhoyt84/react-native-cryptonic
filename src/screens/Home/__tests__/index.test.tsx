import * as React from "react";
import "react-native";
import Home from "../index";
// Note: test renderer must be required after react-native.
import * as renderer from "react-test-renderer";

const navigation = { navigate: jest.fn() };
const loadRecentTradeDataForKeyPair = jest.fn();
const viewingCurrentAssetTrades = "";
const assets = {};

it("renders correctly", () => {
    const tree = renderer.create(<Home assets={assets} loadRecentTradeDataForKeyPair={loadRecentTradeDataForKeyPair} recentTradeData={[]} viewingCurrentAssetTrades={viewingCurrentAssetTrades} loadingAssets={false}/>).toJSON();
    expect(tree).toMatchSnapshot();
});
