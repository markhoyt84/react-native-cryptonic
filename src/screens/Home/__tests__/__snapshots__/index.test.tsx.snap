// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#FBFAFA",
      },
    ]
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "700",
              "marginLeft": undefined,
              "paddingLeft": 4,
              "paddingTop": 1,
              "textAlign": "center",
            }
          }
        >
          Cryptonic Markets
        </Text>
      </View>
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        closeOnRowBeginSwipe={false}
        closeOnRowPress={true}
        closeOnScroll={true}
        directionalDistanceChangeThreshold={2}
        disableLeftSwipe={false}
        disableRightSwipe={false}
        leftOpenValue={0}
        previewFirstRow={false}
        recalculateHiddenLayout={false}
        rightOpenValue={0}
        style={Object {}}
        swipeToOpenPercent={50}
      />
    </View>
  </RCTScrollView>
</View>
`;
